### PR TITLE
[mljar-supervised] Patch predictions for boolean target

### PR DESCRIPTION
**Pull request by PGijsbers**
Models trained with a boolean target would output integer label predictions ([mljar#442](https://github.com/mljar/mljar-supervised/issues/442)). This patch addresses it by specifically looking at this mismatch, and if so converting the target labels back to boolean values. Additionally the probabilities_labels are now explicitly passed, to make sure there is no accidental mismatch.

This issue lead to failure in `kc1`.

### CODE REVIEW

**Comment by PGijsbers** [comment]
@sebhrusen I am a bit confused about the reported failure on `APSFailure`. It reads the job was killed but the specified runtime was [`600 seconds`](https://github.com/openml/automlbenchmark/pull/364/checks?check_run_id=3154568013#step:7:205) and the job was reported to finish in under that time [here](https://github.com/openml/automlbenchmark/pull/364/checks?check_run_id=3154568013#step:7:282). How come the stop was already invoked? It shouldn't include setup time, right?

**Comment by PGijsbers** [comment]
I ran `python runbenchmark.py mljarsupervised_compete validation test -m aws -p 10` successfully with one exception where the time constraint was violated (`bioresponse fold 0`), so I think the integration script is working fine.

**Comment by sebhrusen** [comment]
@PGijsbers I don't know why the `APSFailure` got cancelled after about 496 sec.
The config shows clearly
```
job_timeout_seconds=1200, max_runtime_seconds=600
```
The 2nd is passed to the framework, and the 1st is the total time before everything is aborted (this 1st timeout can include data serialization/deserialization between processes).

I will try to run it locally, but my guess is that it was interrupted by the workflow itself.

**Review by sebhrusen — APPROVED** [review]
looks great, thanks!

**Comment by sebhrusen** [comment]
FYI, to be compared with a worst case scenario: 10 min `max_runtime` + 5 min time overhead for data preparation, all of this limited by the hard limit ( 2x max_runtime = 20 min ).
mljar didn't produce any model after the 10 min, `amlb` tries to interrupt the process "softly" after 15 min (after overhead), and then does a hard interrupt after `20 min`:

```
Running task APSFailure on framework mljarsupervised with config:
TaskConfig(framework='mljarsupervised', framework_params={}, framework_version='0.10.6', type='classification', name='APSFailure', fold=0, metric='auc', metrics=['auc', 'logloss', 'acc', 'balacc'], seed=196968564, job_timeout_seconds=900, max_runtime_seconds=600, cores=4, max_mem_size_mb=3875, min_vol_size_mb=-1, input_dir='/Users/seb/.openml/cache', output_dir='/Users/seb/repos/ml/automlbenchmark/results/mljarsupervised.validation.test.local.20210726T155536', output_predictions_file='/Users/seb/repos/ml/automlbenchmark/results/mljarsupervised.validation.test.local.20210726T155536/predictions/APSFailure/0/predictions.csv', ext={}, type_='binary', output_metadata_file='/Users/seb/repos/ml/automlbenchmark/results/mljarsupervised.validation.test.local.20210726T155536/predictions/APSFailure/0/metadata.json')
Running cmd `/Users/seb/repos/ml/automlbenchmark/frameworks/mljarsupervised/venv/bin/python -W ignore /Users/seb/repos/ml/automlbenchmark/frameworks/mljarsupervised/exec.py`
2021-07-26 17:55:44,742 exec.py INFO
**** mljar-supervised [v0.10.6] ****

2021-07-26 18:10:44,796 amlb.utils.process WARNING Interrupting thread MainThread [ident=4765527552] after 900s timeout.
2021-07-26 18:10:44,859 supervised.preprocessing.eda ERROR There was an issue when running EDA. Interrupting thread MainThread [ident=4765527552] after 900s timeout.
2021-07-26 18:10:48,416 supervised.exceptions ERROR No models produced.
Please check your data or submit a Github issue at https://github.com/mljar/mljar-supervised/issues/new.
2021-07-26 18:10:48,418 frameworks.shared.callee ERROR Interrupting thread MainThread [ident=4765527552] after 900s timeout.
Linear algorithm was disabled.
Interrupting thread MainThread [ident=4576484864] after 1200s timeout.
Interrupting thread MainThread [ident=4576484864] after 1200s timeout.
Interrupting thread MainThread [ident=4576484864] after 1200s timeout.
Traceback (most recent call last):
  File "/Users/seb/repos/ml/automlbenchmark/amlb/benchmark.py", line 512, in run
    meta_result = self.benchmark.framework_module.run(self._dataset, task_config)
  File "/Users/seb/repos/ml/automlbenchmark/frameworks/mljarsupervised/__init__.py", line 26, in run
    input_data=data, dataset=dataset, config=config)
  File "/Users/seb/repos/ml/automlbenchmark/frameworks/shared/caller.py", line 127, in run_in_venv
    _monitor_=monitor
  File "/Users/seb/repos/ml/automlbenchmark/amlb/utils/process.py", line 232, in run_cmd
    preexec_fn=params.preexec_fn)
  File "/Users/seb/repos/ml/automlbenchmark/amlb/utils/process.py", line 64, in run_subprocess
    stdout, stderr = communicate(process, input, timeout=timeout)
  File "/Users/seb/repos/ml/automlbenchmark/amlb/utils/process.py", line 59, in communicate
    return (communicate_fn(process, input=input, timeout=timeout) if communicate_fn
  File "/Users/seb/repos/ml/automlbenchmark/amlb/utils/process.py", line 213, in communicate
    res = h(*args, **kwargs)
  File "/Users/seb/repos/ml/automlbenchmark/amlb/utils/process.py", line 156, in live_output_unix
    ['', '']))
  File "/Users/seb/repos/ml/automlbenchmark/amlb/utils/process.py", line 155, in <lambda>
    process.stderr if process.stderr else 2], activity_timeout),
  File "/Users/seb/repos/ml/automlbenchmark/amlb/utils/process.py", line 141, in read_pipe
    ready, *_ = select.select(pipes, [], [], timeout)
  File "/Users/seb/repos/ml/automlbenchmark/amlb/benchmark.py", line 229, in on_interrupt
    def on_interrupt(*_):
amlb.utils.process.TimeoutError: Interrupting thread MainThread [ident=4576484864] after 1200s timeout.
Loading metadata from `/Users/seb/repos/ml/automlbenchmark/results/mljarsupervised.validation.test.local.20210726T155536/predictions/APSFailure/0/metadata.json`.
Metric scores: { 'acc': nan,
  'app_version': 'dev [fix-mljar, a559f78]',
  'auc': nan,
  'balacc': nan,
  'constraint': 'test',
  'duration': nan,
  'fold': 0,
  'framework': 'mljarsupervised',
  'id': 'openml.org/t/168868',
  'info': 'TimeoutError: Interrupting thread MainThread [ident=4576484864] '
          'after 1200s timeout.',
  'logloss': nan,
  'metric': 'auc',
  'mode': 'local',
  'models_count': nan,
  'params': '',
  'predict_duration': nan,
  'result': nan,
  'seed': 196968564,
  'task': 'APSFailure',
  'training_duration': nan,
  'type': 'binary',
  'utc': '2021-07-26T16:16:38',
  'version': '0.10.6'}
Job `local.validation.test.APSFailure.0.mljarsupervised` executed in 1262.468 seconds.
All jobs executed in 1262.637 seconds.
Processing results for mljarsupervised.validation.test.local.20210726T155536
Scores saved to `/Users/seb/repos/ml/automlbenchmark/results/mljarsupervised.validation.test.local.20210726T155536/scores/mljarsupervised.task_APSFailure.csv`.
Scores saved to `/Users/seb/repos/ml/automlbenchmark/results/mljarsupervised.validation.test.local.20210726T155536/scores/results.csv`.
Scores saved to `/Users/seb/repos/ml/automlbenchmark/results/results.csv`.
Summing up scores for current run:
                 id       task fold       framework constraint metric  duration      seed                                                                                 info
openml.org/t/168868 APSFailure    0 mljarsupervised       test    auc    1262.5 196968564 TimeoutError: Interrupting thread MainThread [ident=4576484864] after 1200s timeout.
```

The error looks very different from what we see in the workflow.
